File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
          	
    	int [] answer = new int[s.length()];
    	
    	
    	for(int i = 0 ; i<s.length(); i++) {
    	answer[i] = -1;	
    	//char c= s.charAt(i);


            // 기준이랑 똑같은 값 인덱스 차이 넣기
    		for(int j = 0; j <i;j++) {
    			if(s.charAt(i)==s.charAt(j)) {
    			
    				answer[i]=  i-j;
    			
    			} 
    		}
    		
    		
    		
    		
    	}
    	
    	
    	
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = new Array(s.length);
    for(var i = 0 ; i < s.length ; i++){
        var index = s.indexOf(s.charAt(i));
        if( i != index){   
            answer[i] = Math.abs(index-i);
            s = s.replace(s.charAt(index), "0");
        } else{
            answer[i] = -1;
        }
    }
    return answer;
}